core/src/main/kotlin/org/evomaster/core/EMConfig.kt

```diff
@@ -550,4 +550,103 @@ class EMConfig {
             "NOTE: this should not cause any tests to fail.")
     var enableBasicAssertions = false
 
+    enum class ResourceSamplingStrategy (val requiredArchive : Boolean = false){
+        NONE,
+        /**
+         * probability for applicable strategy is specified
+         */
+        Customized,
+        /**
+         * probability for applicable strategy is equal
+         */
+        EqualProbability,
+        /**
+         * probability for applicable strategy is derived based on actions
+         */
+        Actions,
+        /**
+         * probability for applicable strategy is adaptive with time
+         */
+        TimeBudgets,
+        /**
+         * probability for applicable strategy is adaptive with performance, i.e., Archive
+         */
+        Archive (true),
+        /**
+         * probability for applicable strategy is adaptive with performance, i.e., Archive
+         */
+        ConArchive (true)
+    }
+
+    @Experimental
+    @Cfg("Specify whether to enable resource-based strategy to sample an individual during search. " +
+            "Note that resource-based sampling is only applicable for REST problem with MIO algorithm.")
+    var resourceSampleStrategy = ResourceSamplingStrategy.NONE
```

Comment:
in the code (`updateProperties()`), we do have checks if configs are not inconsistent. should add check that, if != NONE, then must be REST and MIO

Reply:
thanks, all resource-related configurations are fixed.